go-controller/pkg/ovn/egressip.go

```diff
@@ -169,6 +169,10 @@ func (oc *Controller) reconcileEgressIP(old, new *egressipv1.EgressIP) (err erro
 			}
 			statusToKeep = append(statusToKeep, statusToAdd...)
 		}
+		// Add all assignments which are to be kept to the allocator cache,
+		// allowing us to track all assignments which have been performed and
+		// avoid incorrect future assignments due to a de-synchronized cache.
+		oc.addAllocatorEgressIPAssignments(name, statusToKeep)
```

Comment:
No, not really: `addEgressIPAssignments` was only called if `len(ipsToAdd) > 0`, which would not always be the case if we restart since `ipsToAdd` contain the diff between what needs to be assigned and that which is already assigned. We need to everything existing to the allocator cache though as to keep the state up-to-date. I will add a better commit message though